pandas/tests/indexing/test_loc.py

```diff
@@ -712,6 +712,17 @@ def test_loc_assign_non_ns_datetime(self, unit):
         expected = Series(df.loc[:, "expected"], name=unit)
         tm.assert_series_equal(df.loc[:, unit], expected)
 
+    def test_loc_modify_datetime(self):
+        # GH 28837
+        df_orig = pd.DataFrame.from_dict(
+            {"date": [1485264372711, 1485265925110, 1540215845888, 1540282121025]}
+        )
+        df_orig["date_dt"] = pd.to_datetime(df_orig["date"], unit="ms", cache=True)
+        df_orig.loc[:, "date_dt_cp"] = df_orig.loc[:, "date_dt"]
+        df = df_orig.copy()
+        df.loc[[2, 3], "date_dt_cp"] = df.loc[[2, 3], "date_dt"]
+        tm.assert_series_equal(df.loc[:, "date_dt_cp"], df_orig.loc[:, "date_dt_cp"])
```

Comment:
Could you use a `result/expcted` terminology here? We would like to also compare the entire frame.

```
result = ...
expected = ...
tm.assert_frame_equal(result, equal)
```

Reply:
Changed.